.circleci/config.yml

```diff
@@ -519,38 +553,61 @@ jobs:
                 name: Set Turf Revision
                 command: |
                   sed -i '' 's/turf-swift.git", from: "2.0.0"/turf-swift.git", .revision("<< parameters.turf-revision >>")/' Package.swift
-      # Building and testing are split into 2, with the aim that we'll be able to reuse
-      # the build product and test on multiple simulators
+      - run:
+          name: Checkout xcodebuild deps
+          command: xcodebuild -resolvePackageDependencies -scheme MapboxMaps -derivedDataPath build
       - run:
           name: Building MapboxMaps for simulator
-          command: make build-sdk-for-testing-simulator
-          no_output_timeout: 5m
+          command: |
+            xcodebuild build-for-testing \
+              -scheme MapboxMaps \
+              -sdk iphonesimulator \
+              -configuration << parameters.configuration >> \
+              -destination 'generic/platform=iOS Simulator' \
+              -derivedDataPath build \
+              -enableCodeCoverage YES \
+              COMPILER_INDEX_STORE_ENABLE=NO \
+              ENABLE_TESTABILITY=YES \
+              ONLY_ACTIVE_ARCH=YES
+      - persist_to_workspace:
+          root: build/Build/Products
+          paths:
+            - << parameters.configuration >>-iphonesimulator
+            - "*.xctestrun"
+
+  run-unit-tests:
+    parameters:
+      destination:
+        type: string
+      xcode:
+        type: string
+        default: "13.1.0"
+    macos:
+      xcode: << parameters.xcode >>
+    resource_class: macos.x86.medium.gen2
+    steps:
+      - attach_workspace:
+          at: build
       - run:
-          name: Testing MapboxMaps with simulator
-          command: make test-sdk-without-building-simulator | xcpretty --report junit --output build/reports/junit.xml
-      - store_test_results:
-          path: build/reports
+          name: Run MapboxMaps unit test on << parameters.destination >>  simulator
+          command: |
+            XCTESTRUN_PATH=$(find build -type f -name '*.xctestrun')
+            xcodebuild \
+              test-without-building \
+              -xctestrun "$XCTESTRUN_PATH" \
+              -destination 'platform=iOS Simulator,<< parameters.destination >>' \
+              -enableCodeCoverage YES \
+              -resultBundlePath MapboxMapsTests.xcresult | xcpretty --report junit --output junit.xml
```

Comment:
Can you extract logs to the separate file using `tee` and store them in artifacts